contrib/platform/src/com/sun/jna/platform/win32/Win32VK.java

```diff
@@ -0,0 +1,302 @@
+	 */
+
+	VK_GAMEPAD_A(0xC3, false, 0x0604), // reserved
+	VK_GAMEPAD_B(0xC4, false, 0x0604), // reserved
+	VK_GAMEPAD_X(0xC5, false, 0x0604), // reserved
+	VK_GAMEPAD_Y(0xC6, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_SHOULDER(0xC7, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_SHOULDER(0xC8, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_TRIGGER(0xC9, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_TRIGGER(0xCA, false, 0x0604), // reserved
+	VK_GAMEPAD_DPAD_UP(0xCB, false, 0x0604), // reserved
+	VK_GAMEPAD_DPAD_DOWN(0xCC, false, 0x0604), // reserved
+	VK_GAMEPAD_DPAD_LEFT(0xCD, false, 0x0604), // reserved
+	VK_GAMEPAD_DPAD_RIGHT(0xCE, false, 0x0604), // reserved
+	VK_GAMEPAD_MENU(0xCF, false, 0x0604), // reserved
+	VK_GAMEPAD_VIEW(0xD0, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_THUMBSTICK_BUTTON(0xD1, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_THUMBSTICK_BUTTON(0xD2, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_THUMBSTICK_UP(0xD3, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_THUMBSTICK_DOWN(0xD4, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_THUMBSTICK_RIGHT(0xD5, false, 0x0604), // reserved
+	VK_GAMEPAD_LEFT_THUMBSTICK_LEFT(0xD6, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_THUMBSTICK_UP(0xD7, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_THUMBSTICK_DOWN(0xD8, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_THUMBSTICK_RIGHT(0xD9, false, 0x0604), // reserved
+	VK_GAMEPAD_RIGHT_THUMBSTICK_LEFT(0xDA, false, 0x0604), // reserved
+
+	VK_OEM_4(0xDB, true), // '[{' for US
+	VK_OEM_5(0xDC, true), // '\|' for US
+	VK_OEM_6(0xDD, true), // ']}' for US
+	VK_OEM_7(0xDE, true), // ''"' for US
+	VK_OEM_8(0xDF, true),
+
+	/*
+	 * 0xE0 : reserved
+	 */
+	VK_RESERVED_E0(0xE0, false),
+
+	/*
+	 * Various extended or enhanced keyboards
+	 */
+	VK_OEM_AX(0xE1, true), // 'AX' key on Japanese AX kbd
+	VK_OEM_102(0xE2, true), // "<>" or "\|" on RT 102-key kbd.
+	VK_ICO_HELP(0xE3, true), // Help key on ICO
+	VK_ICO_00(0xE4, false), // 00 key on ICO, produces "00"
+
+	VK_PROCESSKEY(0xE5, true, 0x0400),
+
+	VK_ICO_CLEAR(0xE6, true),
+
+	VK_PACKET(0xE7, true, 0x0500),
+
+	/*
+	 * 0xE8 : unassigned
+	 */
+	VK_UNASSIGNED_E8(0xE8, false),
+
+	/*
+	 * Nokia/Ericsson definitions
+	 */
+	VK_OEM_RESET(0xE9, true), VK_OEM_JUMP(0xEA, true), VK_OEM_PA1(0xEB, true), VK_OEM_PA2(0xEC, true),
+	VK_OEM_PA3(0xED, true), VK_OEM_WSCTRL(0xEE, true), VK_OEM_CUSEL(0xEF, true), VK_OEM_ATTN(0xF0, true),
+	VK_OEM_FINISH(0xF1, true), VK_OEM_COPY(0xF2, true), VK_OEM_AUTO(0xF3, true), VK_OEM_ENLW(0xF4, true),
+	VK_OEM_BACKTAB(0xF5, true),
+
+	VK_ATTN(0xF6, true), VK_CRSEL(0xF7, true), VK_EXSEL(0xF8, true), VK_EREOF(0xF9, true), VK_PLAY(0xFA, true),
+	VK_ZOOM(0xFB, true), VK_NONAME(0xFC, true), VK_PA1(0xFD, true), VK_OEM_CLEAR(0xFE, true),
+	/*
+	 * 0xFF : reserved
+	 */
+	VK_RESERVED_FF(0xFF, false);
+
+	/**
+	 * The code value.
+	 */
+	public final int code;
+	/** This VK may be used to map to a unicode codepoint in a Keyboard Layout. */
+	public final boolean mappable;

```

Comment:
This is based on own investigation. I understand that this might be borderline to include here, but it is such a useful bit of information. I can refactor the code to leave it out if you consider it out-of-policy.

Reply:
I gave this a second thought and went for a feasible alternative. I have taken out mappable member and introduced User32Util.WIN32VK_MAPPABLE, a set of the mappable virtual keys. This makes the enum clean (based purely on header definitions) and keeps the information in a concise way as a useful utility.